react.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+export const usePouchy = (store, options = {}, id = null) => {
+  const [data, setData] = React.useState([]);
+  const refStore = React.useRef(store);
+  // const refOptions = React.useRef(options);
+
+  React.useEffect(() => {
+    if (store.isInitialized) {
+      (async () => {
+        const newData = await refStore.current.getDocuments(options);
+        setData(newData);
+      })();
+
+      return refStore.current.subscribe(newData => {
+        setData(newData);
+      }, options);
+    }
+  }, [store.isInitialized, id]);
```

Comment:
@ajiagahari there's no warning hook dependencies here?

Reply:
@sumartoyo 
the first one I used inside my function, for updating options,
and second, is the declaration on base react function, 

and the last one, yup there is `exhaustive-deps` warning

Reply:
That's actually why I use useRef, to handle the warning. But if we include `options` in the dependencies, the useEffect will always called everytime (if the options object is not memoized).

Reply:
Yes, thats exactly why I didn't include options inside deps
Actualy if you can help me get example code to update ref like props without using dom that would be best solution, I've tried several code, but always failed in updating the ref, 
The new ref always need new declaration, but that would be add another hooks inside component right?
@sumartoyo

Reply:
Yep, you're right. Let me think for a second.

Reply:
I think we can use string representation of `options` for this case.

For example, like this:
```js
export const usePouchy = (store, options = {}) => {
  const [data, setData] = React.useState([]);
  const refStore = React.useRef(store);
  const optionsJson = JSON.stringify(options);
  
  React.useEffect(() => {
    const optionsObj = JSON.parse(optionsJson);
    ...
    const newData = await refStore.current.getDocuments(optionsObj);
    ...
  }, [store.isInitialized, optionsJson]);
```

This way useEffect won't be called if `options` is new object, as long as the contents is the same with previous. And anytime the contents of `options` is changed, useEffect will be called. We don't even need `id` anymore because we don't need to force-call the useEffect.

But it only works if `options` only contains simple data types. For `find` function this is completely okay https://pouchdb.com/api.html#query_index. But for `changes` operation https://pouchdb.com/api.html#changes we can't use `options.filter` function because JSON.parse of a function won't be the function itself.

Reply:
> > @ajiagahari can you explain to me why do you need method `getDocuments`?
> 
> method `fetchData` doesn't have options such as filter, sort, field
> so I made that method just in case method `find` is present, if not it will fallback to method `fetchData`

oops my bad, method `fetchData` have filter options